src/colorizer/ColorizeCanvas.ts

```diff
@@ -185,8 +287,65 @@ export default class ColorizeCanvas {
     this.setUniform("outlierColor", color);
   }
 
-  setHighlightedId(id: number): void {
-    this.setUniform("highlightedId", id);
+  setSelectedTrack(track: Track | null): void {
+    if (this.track && this.track?.trackId === track?.trackId) {
+      return;
+    }
+    this.track = track;
+    if (!track || !track.centroids || track.centroids.length === 0 || !this.frameResolution) {
+      return;
+    }
+    // Make a new array of the centroid positions, normalizing to screen space.
+    // Points is in 3D while centroids are pairs of 2D coordinates in a 1D array
+    this.points = new Float32Array(track.length() * 3);
+    for (let i = 0; i < track.length(); i++) {
+      // Screen space is in [-1, 1] range.
+      this.points[3 * i + 0] = (track.centroids[2 * i] / (4.0 * this.frameResolution.x)) * 2.0 - 1.0;
+      this.points[3 * i + 1] = -((track.centroids[2 * i + 1] / (4.0 * this.frameResolution.y)) * 2.0 - 1.0);
```

Comment:
the only remaining thing I would change is to do this part of the math `/ (4.0 * this.frameResolution.x)) * 2.0 - 1.0` inside the shader instead of here.  
At this place in the code, you should just copy the values from track.centroids into this.points without having to modify the values.   
This is not a deal breaker but what it accomplishes is to take this bit of math that's applied to every value in this linear loop and instead run it on the gpu where it is parallelized over batches of points.  It also gives you the opportunity to optimize copying the raw values.